### PR TITLE
Added new template: show interface link on cisco ios

### PR DESCRIPTION
**Pull request by jversch6**
This has different outputs on several platforms and often doesn't return some fields at all, hence the multitude of configurations.

### CODE REVIEW

File: ntc_templates/templates/cisco_ios_show_interface_link.textfsm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ^\s*Port\s+Name\s+Down Time\s+Up Time\s*$$ -> UpTime
+  ^\s*$$
+  ^. -> Error
+
+DownSince
+  ^\s*${PORT}\s+${DOWNTIME}\s*$$ -> Record
+  ^\s*${PORT}\s+${DOWNTIME}\s*${SINCE}\s*$$ -> Record
+  ^\s*${PORT}\s+${NAME}\s*${DOWNTIME}\s*$$ -> Record
+  ^\s*${PORT}\s+${NAME}\s*${DOWNTIME}\s+${SINCE}\s*$$ -> Record
+
```
Comment (itdependsnetworks) [inline ntc_templates/templates/cisco_ios_show_interface_link.textfsm:24]: Can you add:

```
  ^\s*$$
  ^. -> Error
```

File: ntc_templates/templates/cisco_ios_show_interface_link.textfsm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+UpTime
+  ^\s*${PORT}\s+${NAME}\s+00:00:00\s+${UPTIME}?\s*$$ -> Record
+  ^\s*${PORT}\s+${DOWNTIME}\s+${UPTIME}?\s*$$ -> Record
+  ^\s*${PORT}\s+${NAME}\s+${DOWNTIME}\s*$$ -> Record
```
Comment (itdependsnetworks) [inline ntc_templates/templates/cisco_ios_show_interface_link.textfsm:28]: Can you add:

```
  ^\s*$$
  ^. -> Error
```

File: tests/cisco_ios/show_interface_link/cisco_ios_show_interface_link2.raw [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Port    Name               Down Time        Down Since
+Gi1/1                      37 weeks, 3 days, 5 hours, 52 minutes 18 secs      08:07:10  Mon Feb 14 2022
+Gi1/6                      15 weeks, 11 minutes 57 secs      13:47:31  Thu Jul 21 2022
+Gi1/10   Fitness           10 weeks, 5 days, 20 hours, 26 minutes 38 secs      17:32:50  Fri Aug 19 2022
```
Comment (itdependsnetworks) [inline tests/cisco_ios/show_interface_link/cisco_ios_show_interface_link2.raw:4]: Is it possible for their to be spaces here? e.g. `ICE Port`

File: ntc_templates/templates/cisco_ios_show_interface_link.textfsm (hunk 2 above)
Comment (jversch6) [reply ntc_templates/templates/cisco_ios_show_interface_link.textfsm]: Added

File: tests/cisco_ios/show_interface_link/cisco_ios_show_interface_link2.raw (hunk 3 above)
Comment (jversch6) [reply tests/cisco_ios/show_interface_link/cisco_ios_show_interface_link2.raw:4]: Added test case and cleaned up some redundant lines.

**Comment by jmcgill298** [comment]
@jversch6 thanks for including the Error line, but I think the templates are not matching all of the output, and the tests are failing.

**Comment by jversch6** [comment]
@jmcgill298 Sorry, apparently I hadn't committed the .raw file . It's showing green now!

**Comment by jvanderaa** [comment]
@itdependsnetworks @jmcgill298 we good here?

**Comment by jmcgill298** [comment]
Thanks @jversch6

**Pull request merged**